src/cape_ros_node.cpp

```diff
@@ -0,0 +1,337 @@
+/*
+ */
+
+#include <ros/ros.h>
+#include <sensor_msgs/CameraInfo.h>
+#include <sensor_msgs/Image.h>
+#include <cape_ros/Planes.h>
+#include <cv_bridge/cv_bridge.h>
+#include <cape_ros/cape.h>
+#include <opencv2/opencv.hpp>
+#include <cape_ros/cape_ros_node.h>
+
+std::array<uint8_t, 3> hsvToRgb(const double h, const double s, const double v)
+{
+  const auto c = v * s;
+  const auto h_prime = fmod(h / 60.0, 6);
+  const auto x = c * (1 - std::abs(fmod(h_prime, 2) - 1));
+  const auto m = v - c;
+
+  const auto v1 = static_cast<uint8_t>(255 * (c + m));
+  const auto v2 = static_cast<uint8_t>(255 * (x + m));
+  const auto v3 = static_cast<uint8_t>(255 * m);
+
+  if (h_prime < 1)
+  {
+    return { v1, v2, v3 };
+  }
+  if (h_prime < 2)
+  {
+    return { v2, v1, v3 };
+  }
+  if (h_prime < 3)
+  {
+    return { v3, v1, v2 };
+  }
+  if (h_prime < 4)
+  {
+    return { v3, v2, v1 };
+  }
+  if (h_prime < 5)
+  {
+    return { v2, v3, v1 };
+  }
+  if (h_prime < 6)
+  {
+    return { v1, v3, v2 };
+  }
+
+  return { v3, v3, v3 };
+}
+
+CapeRosNode::CapeRosNode(ros::NodeHandle nh)
+  : nh_(nh)
+  , cos_angle_max_(std::cos(M_PI / 30))
+  , max_merge_dist_(50.0)
+  , depth_cutoff_(5.0)
+  , cylinder_detection_(false)
+  , patch_size_(20)
+{
+  depth_sub = nh.subscribe("depth_in", 2, &CapeRosNode::depthCallback, this);
+  intensity_sub = nh.subscribe("ir_in", 2, &CapeRosNode::intensityCallback, this);
+  planes_pub_ = nh.advertise<cape_ros::Planes>("planes", 1);
+  image_overlay_pub_ = nh.advertise<sensor_msgs::Image>("overlay_image", 1);
+
+  constexpr auto deg_to_rad = M_PI / 180;
+
+  auto max_angle = 6.0; // degrees
+  nh_.param("max_angle", max_angle, max_angle);
+  cos_angle_max_ = std::cos(max_angle * deg_to_rad);
+  nh_.param("max_merge_distance", max_merge_dist_, max_merge_dist_);
+  nh_.param("patch_size", patch_size_, patch_size_);
+  nh_.param("depth_cuttoff", depth_cutoff_, depth_cutoff_);
```

Comment:
Is it ok if we add `cylinder_detection_ `to expose the params?

Reply:
We do not have a msg to publish cylinders yet, so it is pointless to expose it now